### DISCUSSION TITLE
Does calling trainer.fit() reset model weights?

### DISCUSSION THREAD

**Discussion by GV1028** [Lightning-AI/pytorch-lightning#9725] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hi, 

I am currently training multiple models and am trying to update model weights outside the trainer.fit() function as well. The code looks something like this:
```
for i in range(10):
    trainer.fit(model, dl)  
    model.update_weights() #Logic to change weights 
```
What is the behavior of trainer.fit() here? Will it start the training with the current model weights or reinitialize it every time?

**Comment by rohitgr7** [comment]
trainer doesn't know how to reinitialize the weights, so it won't. it will just continue using the weights you passed.
